app/models/participant_profile_state.rb

```diff
@@ -27,4 +13,30 @@ def valid?(_context = nil)
   }
 
   scope :most_recent, -> { order("created_at desc").limit(1) }
+
+private
+
+  def withdrawal
+    return unless current_state
+
+    errors.add(:base, I18n.t(:invalid_withdrawal)) if current_state.state == self.class.states[:withdrawn]
```

Comment:
`errors.add(:base, I18n.t(:invalid_withdrawal)) if current_state.withdrawn?`?

Likewise for other conditionals in this file :)